### PR TITLE
fix: cuts time to get an app export

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3935)

- [x] This change addresses part of the issue
## Description
This should speed up the time to get an application export pretty significantly. 
This should also keep our memory footprint lower than it used to be (in line with the recent change for better memory management)

## How Can This Be Tested/Reviewed?
- Easiest way to check this is to use a listing with >9k applications tied to it
- request an application export for that listing
- you should notice the export shows up much faster and importantly the data in that csv **should **not** have changed
- you should also see the memory footprint remain low (around 200MB on my machine locally)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/src/services/application-csv-export.service.ts [hunk 1]
```diff
@@ -131,131 +141,120 @@ export class ApplicationCsvExporterService
...
+                              race: true,
+                            },
+                          }
+                        : false,
+                    },
+                    where: {
+                      listingId: queryParams.listingId,
+                      deletedAt: null,
+                    },
+                    skip: i,
```
Comment (emilyjablonski) [inline api/src/services/application-csv-export.service.ts:172]: Curious why this changed from (i * num_to_paginate_by) to just i

Comment (YazeedLoonat) [reply api/src/services/application-csv-export.service.ts:172]: ah yeah that has to do with how I was iterating earlier 
in the for loop I was trying to count by batch number basically (0,1,2,..n) but I realized its easier to just count by where we are in the batch (0, 500, 1k, 1.5k, n) instead so we don't need to figure out how many to skip ourselves anymore

File: api/src/services/application-csv-export.service.ts [hunk 2]
```diff
@@ -131,131 +141,120 @@ export class ApplicationCsvExporterService
...
+                let row = '';
+                paginatedApplications.forEach((app) => {
+                  let preferences: ApplicationMultiselectQuestion[];
+                  csvHeaders.forEach((header, index) => {
+                    let multiselectQuestionValue = false;
+                    let parsePreference = false;
+                    let value = header.path.split('.').reduce((acc, curr) => {
+                      // return preference/program as value for the format function to accept
+                      if (multiselectQuestionValue) {
+                        return acc;
```
Comment (mcgarrye) [inline api/src/services/application-csv-export.service.ts:185]: How does line 184 ever evaluate to true?

Comment (YazeedLoonat) [reply api/src/services/application-csv-export.service.ts:185]: as we iterate through in the `reduce` first `parsePreference` can become true, and then if that becomes true `multiselectQuestionValue` becomes true

Comment (YazeedLoonat) [reply api/src/services/application-csv-export.service.ts:185]: as we iterate through with the `reduce()` first `parsePreference` (line 206) can become true, and then later `multiselectQuestionValue` (200) can become true

**Review by mcgarrye — APPROVED** [review]
CSV downloaded in roughly 6s

**Comment by ColinBuyck** [comment]
@YazeedLoonat Do you think there's room in this PR to expand seeding functionality to better support testing things at scale? I'm thinking about like a yarn setup 10000 that would seed 10,000 applications on one of the listing so we could more regularly test performance and ensure any other application/export changes keep response times reasonable (<10 sec)

**Comment by YazeedLoonat** [comment]
Hey @ColinBuyck I'm not sure if adding that would add much

The 10 seconds isn't really a constraint outside of netlify (infrastructural constraint) 

and I think a better/more sustainable way to test this is to use an existing dataset with have access too (production data for example) and anonymizing that and setting that up in a test env that we can qa things against (like setting that data in staging)

where we also have similar enough infrastructure to production to catch timeouts and stuff like that

**Review by ColinBuyck — APPROVED** [review]
LGTM 🧷 I saw a 10 second improvement on one with 10k+ applications, very exciting! One nit but in the future, could you explain **why** changes improve performance in the PR description rather than just noting the areas of improvement. Also for documentation sake, the testing conversation above boiled down to this work per offline discussion #3995

**Pull request merged**